pyclaragenomics/build_clara_genomics_analysis.py

```diff
@@ -0,0 +1,88 @@
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+import argparse
+import os.path
+import os
+import subprocess
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='build & install Clara Genomics Analysis SDK.')
+    parser.add_argument('--build_output_folder',
+                        required=False,
+                        default=None,
+                        help="Choose an output folder for building")
+    return parser.parse_args()
+
+
+class CMakeWrapper:
+    """Class to encapsulate building a CMake project."""
+
+    def __init__(self, cmake_root_dir, cmake_build_path="cmake_build", cmake_extra_args=""):
+        """
+        Class constructor.
+
+        Args:
+            cmake_root_dir : Root directory of CMake project
+            cmake_install_dir : Install location for CMake project
+            cmake_extra_args : Extra string arguments to be passed to CMake during setup
+        """
+        self.cmake_root_dir = os.path.abspath(cmake_root_dir)
+        self.build_path = os.path.abspath(cmake_build_path)
+        self.cmake_install_dir = os.path.join(self.build_path, "install")
+        self.cmake_extra_args = cmake_extra_args
+        self.cuda_toolkit_root_dir = os.environ.get("CUDA_TOOLKIT_ROOT_DIR")
+
+    def run_cmake_cmd(self):
+        cmake_args = ['-DCMAKE_INSTALL_PREFIX=' + self.cmake_install_dir,
+                      '-DCMAKE_BUILD_TYPE=' + 'Release',
+                      '-DCMAKE_INSTALL_RPATH=' + os.path.join(self.cmake_install_dir, "lib")]
+        cmake_args += [self.cmake_extra_args]
+
+        if self.cuda_toolkit_root_dir:
+            cmake_args += ["-DCUDA_TOOLKIT_ROOT_DIR=%s" % self.cuda_toolkit_root_dir]
+
+        if not os.path.exists(self.build_path):
+            os.makedirs(self.build_path)
+
+        subprocess.check_call(['cmake', self.cmake_root_dir] + cmake_args, cwd=self.build_path)
+
+    def run_build_cmd(self):
+        build_args = ['--', '-j16', 'install']
+        subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_path)
+
+    def build(self):
+        self.run_cmake_cmd()
+        self.run_build_cmd()
+
+    def get_installed_path(self, component=""):
+        installed_path = os.path.abspath(os.path.join(self.cmake_install_dir, component))
+        if not os.path.exists(installed_path):
+            raise RuntimeError("No valid path for requested component exists")
+        return installed_path
+
+
+if __name__ == "__main__":
+    args = parse_arguments()
+
+    # Initialize builds.
+    pycga_directory = os.path.dirname(os.path.realpath(__file__))
+    cmake_root_dir = os.path.dirname(pycga_directory)
+    output_folder = \
+        args.build_output_folder if args.build_output_folder else os.path.join(pycga_directory, "cga_build")
```

Comment:
we can make the default of `build_output_folder` as `pycga_dir/cga_build` so we won't need this check here